org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/server/wizard/ServerConnectionWizard.java

```diff
@@ -0,0 +1,220 @@
+  }
+
+  @Override
+  public IWizardPage getStartingPage() {
+    if (!model.isEdit()) {
+      return connectionTypeWizardPage;
+    }
+    return firstPageAfterConnectionType();
+  }
+
+  @Override
+  public void addPages() {
+    if (!model.isEdit()) {
+      addPage(connectionTypeWizardPage);
+      addPage(serverIdPage);
+    }
+    addPage(urlPage);
+    addPage(authMethodPage);
+    addPage(credentialsPage);
+    addPage(tokenPage);
+    addPage(orgPage);
+    addPage(endPage);
+  }
+
+  @Override
+  public IWizardPage getNextPage(IWizardPage page) {
+    if (page == connectionTypeWizardPage) {
+      return firstPageAfterConnectionType();
+    }
+    if (page == urlPage) {
+      return authMethodPage;
+    }
+    if (page == authMethodPage) {
+      return model.getAuthMethod() == AuthMethod.PASSWORD ? credentialsPage : tokenPage;
+    }
+    if (page == credentialsPage || page == tokenPage) {
+      return orgPage;
+    }
+    if (page == orgPage) {
+      return model.isEdit() ? endPage : serverIdPage;
+    }
+    if (page == serverIdPage) {
+      return endPage;
+    }
+    return null;
+  }
+
+  @Override
+  public IWizardPage getPreviousPage(IWizardPage page) {
+    // This method is only used for the first page of a wizard,
+    // because every following page remember the previous one on its own
+    return null;
+  }
+
+  private IWizardPage firstPageAfterConnectionType() {
+    // Skip URL and auth method page if SonarCloud
+    return model.getConnectionType() == ConnectionType.SONARCLOUD ? tokenPage : urlPage;
+  }
+
+  @Override
+  public boolean canFinish() {
+    IWizardPage currentPage = getContainer().getCurrentPage();
+    return currentPage == endPage && super.canFinish();
+  }
+
+  @Override
+  public boolean performFinish() {
+    if (model.isEdit() && !testConnection(model.getOrganization())) {
+      return false;
+    }
+    IServer server;
+    if (model.isEdit()) {
+      editedServer.updateConfig(model.getServerUrl(), model.getOrganization(), model.getUsername(), model.getPassword());
+      server = editedServer;
+    } else {
+      server = SonarLintCorePlugin.getServersManager().create(model.getServerId(), model.getServerUrl(), model.getOrganization(), model.getUsername(), model.getPassword());
+      SonarLintCorePlugin.getServersManager().addServer(server, model.getUsername(), model.getPassword());
+    }
+
+    Job j = new ServerUpdateJob(server);
+    j.addJobChangeListener(new JobChangeAdapter() {
+      @Override
+      public void done(IJobChangeEvent event) {
+        if (event.getResult().isOK()) {
+          JobUtils.scheduleAnalysisOfOpenFilesInBoundProjects(server, TriggerType.BINDING_CHANGE);
+        }
+      }
+    });
+    j.schedule();
+    return true;
+  }
+
+  public boolean beforeNextPressed() {
+    IWizardPage currentPage = getContainer().getCurrentPage();
+    if (currentPage == credentialsPage || currentPage == tokenPage) {
+      if (!testConnection(null)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1066)